portality/static/js/application_form.js

```diff
@@ -866,3 +866,80 @@ window.Parsley.addValidator("notIf", {
     },
     priority: 1
 });
+
+
+///////////////////////////////////////////////////////////////
+// workaround to allow underscores on the parsley type=url validator
+///////////////////////////////////////////////////////////////
+
+// These are the parsley default type testers, sourced from
+// https://github.com/guillaumepotier/Parsley.js/blob/master/src/parsley/validator_registry.js#L15
+// but with our own custom URL validator
+doaj.af.typeTesters =  {
+  email: /^((([a-zA-Z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+(\.([a-zA-Z]|\d|[!#\$%&'\*\+\-\/=\?\^_`{\|}~]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])+)*)|((\x22)((((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(([\x01-\x08\x0b\x0c\x0e-\x1f\x7f]|\x21|[\x23-\x5b]|[\x5d-\x7e]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(\\([\x01-\x09\x0b\x0c\x0d-\x7f]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))))*(((\x20|\x09)*(\x0d\x0a))?(\x20|\x09)+)?(\x22)))@((([a-zA-Z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])|(([a-zA-Z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-zA-Z]|\d|-|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-zA-Z]|\d|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])))\.)+(([a-zA-Z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])([a-zA-Z]|\d|-|_|~|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF])*([a-zA-Z]|[\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]))$/,
+
+  // Follow https://www.w3.org/TR/html5/infrastructure.html#floating-point-numbers
+  number: /^-?(\d*\.)?\d+(e[-+]?\d+)?$/i,
+
+  integer: /^-?\d+$/,
+
+  digits: /^\d+$/,
+
+  alphanum: /^\w+$/i,
+
+  date: {
+    test: value => Utils.parse.date(value) !== null
+  },
+
+  url: new RegExp(
+      "^https?://([^/:]+\.[a-z]{2,63}|([0-9]{1,3}\.){3}[0-9]{1,3})(:[0-9]+)?(\/.*)?$"
```

Comment:
```suggestion
      url: /^https?:\/\/([^/:]+\.[a-z]{2,63}|([0-9]{1,3}\.){3}[0-9]{1,3})(:[0-9]+)?(\/.*)?$/
```
Since this regex isn't changing, I think it's preferred to use the literal notation like those above

Reply:
because of the extra complexity that escaping the slashes adds to this, I'm going to disagree with you on this one - while the style might be preferred, the readability is better this way round.